llvm/lib/Analysis/InstructionSimplify.cpp

```diff
@@ -3103,6 +3103,54 @@ static Value *simplifyICmpWithConstant(CmpInst::Predicate Pred, Value *LHS,
   return nullptr;
 }
 
+/// Get values V_i such that V uge V_i (Greater) or V ule V_i (!Greater).
+static void getUnsignedMonotonicValues(SmallPtrSetImpl<Value *> &Res, Value *V,
```

Comment:
Prefer ptr to output reference.

Reply:
On that note, however, do you actually need to collect the values? Based on the usage of just simplifying booleans, can't you just pass a query (greater, greater_eq, less_eq, less) and return a bool if its provable?

Reply:
Values are collected into the set in recursive calls, so this can't be a return value.

Reply:
err, yeah realized err in my prior comment (deleted shortly after for context).